src/blockly/generators/propc/gpio.js

```diff
@@ -1834,50 +1844,46 @@ Blockly.Blocks.sd_file_pointer = {
     init: function () {
         this.setTooltip(Blockly.MSG_SD_FILE_POINTER_TOOLTIP);
         this.setColour(colorPalette.getColor('output'));
-        this.appendValueInput("FP")
-                .setCheck("Number")
-                .appendField("SD file")
-                .appendField(new Blockly.FieldDropdown([
-                    ["set", "set"],
-                    ["get", "get"]
-                ], function (m) {
-                    this.sourceBlock_.setSdMode(m);
-                }), "MODE")
-                .appendField("pointer = ");
+        this.setSdMode('set');
         this.setInputsInline(false);
         this.setPreviousStatement(true, "Block");
         this.setNextStatement(true, null);
     },
     mutationToDom: Blockly.Blocks['sd_read'].mutationToDom,
     domToMutation: Blockly.Blocks['sd_read'].domToMutation,
-    setSdMode: function (m) {
+    setSdMode: function (mode) {
         if(this.getInput('FP')) {
             this.removeInput('FP');
         }
-        var meq = '';
-        if (m === 'set') {
-            meq = ' = ';
-            this.setOutput(false);
+        if (mode === 'set') {
             this.appendValueInput('FP')
-                    .setCheck("Number");
+                    .setCheck("Number")
+                    .appendField("SD file")
+                    .appendField(new Blockly.FieldDropdown([
+                        ["set", "set"],
+                        ["get", "get"]
+                    ], function (blockMode) {
+                        this.sourceBlock_.setSdMode(blockMode);
+                    }), "MODE")
+                    .appendField("pointer = ");
+            this.setOutput(false);
+            this.setPreviousStatement(true, "Block");
+            this.setNextStatement(true, null);
         } else {
-            this.setOutput(true, 'Number');
             this.appendDummyInput('FP');
+            this.getInput('FP')
+                    .appendField("SD file")
+                    .appendField(new Blockly.FieldDropdown([
+                        ["get", "get"],
+                        ["set", "set"]
+                    ], function (blockMode) {
+                        this.sourceBlock_.setSdMode(blockMode);
+                    }), "MODE")
+                    .appendField("pointer");
+            this.setPreviousStatement(false, "Block");
+            this.setNextStatement(false, null);
+            this.setOutput(true, 'Number');
         }
-        this.getInput('FP')
-                .appendField("SD file")
-                .appendField(new Blockly.FieldDropdown([
-                    ["set", "set"],
-                    ["get", "get"]
-                ], function (m) {
-                    this.sourceBlock_.setSdMode(m);
-                }), "MODE")
-                .appendField("pointer" + meq);
-        this.setFieldValue(m, 'MODE');
```

Comment:
Line 1876 (above) is the culprit for these types of errors.  This is actually the same issue we had with the comment block.  It seems that when the Blockly core was updated, a new race condition was introduced.
This caused the change handler for the dropdown menu to get called even when the value of the field is changed programmatically.  This then set up an infinite loop - the menu was changed, so the change handler was called, which changed the menu, which called the handler...and on and on and on.

The solution is to rebuild the menu is a way that doesn't require the menu to the be set to a new value later.  This means that instead of setting the menu, making the desired option the **first** option.